batdata/data.py

```diff
@@ -234,6 +240,43 @@ def from_batdata_hdf(cls, path_or_buf: Union[str, Path, HDFStore], subsets: Opti
 
         return cls(**data, metadata=metadata)
 
+    @staticmethod
+    def inspect_batdata_hdf(path: Union[str, Path]) -> tuple[BatteryMetadata, Set[Optional[str]]]:
+        """Extract the battery data and the names of cells contained within an HDF5 file
+
+        Args:
+            path: Path to the HDF5 file
+        Returns:
+            - Metadata from this file
+            - List of names of batteries stored within the file
+        """
+
+        with h5py.File(path, 'r') as f:
+            metadata = BatteryMetadata.model_validate_json(f.attrs['metadata'])
+
+            # Get the names by gathering all names before the "-" in group names
+            names = set()
+            for key in f.keys():
+                if '-' not in key:  # From the "default" group
+                    names.add(None)
+                name, _ = key.rsplit("-", 1)
+                names.add(name)
+            return metadata, names
+
+    @staticmethod
+    def get_metadata_from_hdf5(path: Union[str, Path]) -> BatteryMetadata:
```

Comment:
I'm a little confused... Is this a different function from what it used to be?

Reply:
Do you mean different than the old method for reading the metadata from an HDF file?

Reply:
Yeah, comparing the additions and deletions on Github it appears they are the same function? Maybe just moved to a different place in the file?

Reply:
Correct. It just moved